first-run: cold 'make all' fails 'invalid reference format' — .env.example BROWSER_IMAGE uses ${IMAGE_TAG}, unexpanded by the Makefile pull

## Where we are (v0.12.14 witness deployment, fresh Ubuntu 24.04)

Cold first-run per README: clone at `v0.12.14` → `make all`. Every service image pulls fine, then:

```
docker.io/vexaai/v012-agent-worker:v0.12.14
invalid reference format
make[1]: *** [Makefile:74: up] Error 1
```

Cause: `.env.example:33` ships `BROWSER_IMAGE=vexaai/vexa-bot:${IMAGE_TAG}`. Docker-compose expands that fine in yml context, but `deploy/compose/Makefile`'s spawn-image pull reads it with grep/cut (`_bi=$(grep '^BROWSER_IMAGE=' .env ...)`) and hands the **literal** `vexaai/vexa-bot:${IMAGE_TAG}` to `docker pull` → invalid reference. The `case "$$_bi" in vexaai/*)` guard matches the literal so the pull is always attempted. **Every cold first-run breaks** — this is a first-run lane blocker on the current release.

## Prepared solution + forks

- Preferred: expand in the Makefile snippet — substitute \${IMAGE_TAG} with the already-computed `_tag` before pulling (one sed), keeping the parameterized .env.example.
- Fork: ship `.env.example` with the literal default (`vexaai/vexa-bot:v012`) — loses the single-knob coupling.

## Acceptance

- [x] Fresh clone + `make all` on a clean host completes to the API-key print with stock .env.example (both IMAGE_TAG=v012 default and an explicit pin).
- [x] Negative control: a genuinely custom BROWSER_IMAGE literal still pulls verbatim.

Workaround used in the witness run: set the literal in .env.

Origin: v0.12.14 release witness provisioning, 2026-07-19.

--- a/deploy/compose/Makefile
+++ b/deploy/compose/Makefile
@@ -71,10 +71,11 @@ up:
 	@$(MAKE) --no-print-directory _preflight
 	@[ -f .env ] || { echo "→ no .env found; seeding from .env.example"; cp .env.example .env; }
 	$(COMPOSE) pull
-	@_tag=$$(grep -E '^IMAGE_TAG=' .env | cut -d= -f2- | tr -d '[:space:]'); _tag=$${_tag:-v012}; \
+	@_tag=$$(grep -E '^IMAGE_TAG=' .env | head -1 | cut -d= -f2- | tr -d '[:space:]'); _tag=$${_tag:-v012}; \
 	 echo "→ pulling spawn-time images (agent-worker is a build-only compose profile pull skips)"; \
 	 docker pull "vexaai/v012-agent-worker:$$_tag"; \
 	 _bi=$$(grep -E '^BROWSER_IMAGE=' .env | head -1 | cut -d= -f2- | sed -E 's/[[:space:]]+#.*$$//; s/[[:space:]]+$$//'); \
+	 _bi=$$(printf '%s' "$$_bi" | sed "s/\$${IMAGE_TAG}/$$_tag/; s/\$$IMAGE_TAG/$$_tag/"); \
 	 case "$$_bi" in vexaai/*) docker pull "$$_bi";; esac
 	$(COMPOSE) up -d --no-build
 	@$(MAKE) --no-print-directory _bot_check

--- a/docs/changelog.d/812-first-run-browser-image.md
+++ b/docs/changelog.d/812-first-run-browser-image.md
@@ -1,0 +1,5 @@
+- **Cold first-run no longer dies on `invalid reference format` (#812).** Stock `.env.example` ships
+  `BROWSER_IMAGE=vexaai/vexa-bot:${IMAGE_TAG}` — docker-compose expands it, but `make all`'s
+  spawn-image pull read the literal string and handed `docker pull` an invalid reference, breaking
+  every fresh install at the last step. The pull now resolves `${IMAGE_TAG}` (and hardens the tag
+  read against duplicate lines); a genuinely custom literal `BROWSER_IMAGE` still pulls verbatim.

